### PR TITLE
test: clean up and augment Tray module spec

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

This PR cleans up the formatting for and adds some new specs for the Tray module.

Tests have been added for `get/set ignoreDoubleClickEvents` as well as for `tray.getBounds()`.

cc @miniak @nornagon @MarshallOfSound 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

File: spec-main/api-tray-spec.ts [hunk 1]
```diff
@@ -12,17 +12,30 @@ describe('tray module', () => {
     tray = null as any
   })
 
-  describe('tray.setContextMenu', () => {
-    afterEach(() => {
-      tray.destroy()
+  describe('tray get/set ignoreDoubleClickEvents', () => {
```
Comment (nornagon) [inline spec-main/api-tray-spec.ts]: please use `ifdescribe` as defined in `spec-helpers.ts`

File: spec-main/api-tray-spec.ts [hunk 2]
```diff
@@ -12,17 +12,30 @@ describe('tray module', () => {
...
+    before(function () {
+      if (process.platform !== 'darwin') this.skip()
     })
 
-    it('accepts menu instance', () => {
-      tray.setContextMenu(new Menu())
+    it('returns false by default', () => {
+      const ignored = tray.getIgnoreDoubleClickEvents()
+      expect(ignored).to.be.false
```
Comment (nornagon) [inline spec-main/api-tray-spec.ts]: this doesn't assert anything, use
```suggestion
      expect(ignored).to.be.false('ignored')
```

File: spec-main/api-tray-spec.ts [hunk 3]
```diff
@@ -12,17 +12,30 @@ describe('tray module', () => {
...
+    })
+
+    it('can be set to true', () => {
+      tray.setIgnoreDoubleClickEvents(true)
+
+      const ignored = tray.getIgnoreDoubleClickEvents()
+      expect(ignored).to.be.true
```
Comment (nornagon) [inline spec-main/api-tray-spec.ts]: same here

File: spec-main/api-tray-spec.ts [hunk 4]
```diff
@@ -35,49 +48,54 @@ describe('tray module', () => {
...
 
-  describe('tray.popUpContextMenu', () => {
-    afterEach(() => {
-      tray.destroy()
-    })
-
-    before(function () {
+  describe('tray.popUpContextMenu()', () => {
+    it('can be called when menu is showing', function (done) {
       if (process.platform !== 'win32') this.skip()
```
Comment (nornagon) [inline spec-main/api-tray-spec.ts]: while you're here, refactor to `ifit`?

File: spec-main/api-tray-spec.ts [hunk 5]
```diff
@@ -35,49 +48,54 @@ describe('tray module', () => {
...
+      tray.destroy()
     })
   })
 
-  describe('tray.setImage', () => {
+  describe('tray.getBounds()', () => {
+    afterEach(() => { tray.destroy() })
+
+    it ('returns a bounds object', function () {
+      if (process.platform === 'linux') this.skip()
```
Comment (nornagon) [inline spec-main/api-tray-spec.ts]: `ifit`

File: spec-main/api-tray-spec.ts [hunk 6]
```diff
@@ -35,49 +48,54 @@ describe('tray module', () => {
...
     it('accepts empty image', () => {
       tray.setPressedImage(nativeImage.createEmpty())
 
       tray.destroy()
     })
   })
 
-  describe('tray title get/set', () => {
+  describe('tray get/set title', () => {
     before(function () {
```
Comment (nornagon) [inline spec-main/api-tray-spec.ts]: refactor to use `ifdescribe`?

File: spec-main/api-tray-spec.ts [hunk 7]
```diff
@@ -12,17 +13,26 @@ describe('tray module', () => {
...
+  })
 
-    it('accepts null', () => {
-      tray.setContextMenu(null)
+  describe('tray.setContextMenu(menu)', () => {
+    it('accepts both null and Menu as parameters', () => {
+      expect(() => { tray.setContextMenu(new Menu()) }).to.not.throw()
+      expect(() => { tray.setContextMenu(null) }).to.not.throw()
+
+      tray.destroy()
```
Comment (nornagon) [inline spec-main/api-tray-spec.ts]: is it ok for this to not be called if one of the `expect`s above fails? if so, maybe we can just not call it at all?

File: spec-main/api-tray-spec.ts [hunk 8]
```diff
@@ -35,49 +45,45 @@ describe('tray module', () => {
...
-      if (process.platform !== 'win32') this.skip()
-    })
-
-    it('can be called when menu is showing', (done) => {
+  describe('tray.popUpContextMenu()', () => {
+    ifit(process.platform === 'win32')('can be called when menu is showing', function (done) {
       tray.setContextMenu(Menu.buildFromTemplate([{ label: 'Test' }]))
       setTimeout(() => {
         tray.popUpContextMenu()
+        tray.destroy()
```
Comment (nornagon) [inline spec-main/api-tray-spec.ts]: same here

File: spec-main/api-tray-spec.ts [hunk 9]
```diff
@@ -35,49 +45,45 @@ describe('tray module', () => {
...
         done()
       })
       tray.popUpContextMenu()
     })
   })
 
-  describe('tray.setImage', () => {
+  describe('tray.getBounds()', () => {
+    afterEach(() => { tray.destroy() })
```
Comment (nornagon) [inline spec-main/api-tray-spec.ts]: ... here to. let's be consistent about whether or not we destroy the tray after a test.

**Comment by codebytere** [comment]
@nornagon i moved all afterEach calls to the top-level describe for consistency, given that i think it's in our best interest to destroy and full re-create a tray for each spec.

**Pull request merged**